x-pack/plugins/ai_infra/llm_tasks/server/tasks/retrieve_documentation/retrieve_documentation.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { OutputAPI } from '@kbn/inference-plugin/common/output';
+import type { ProductDocSearchAPI } from '@kbn/product-doc-base-plugin/server';
+import { truncate, count as countTokens } from '../../utils/tokens';
+import type { RetrieveDocumentationAPI } from './types';
+import { summarizeDocument } from './summarize_document';
+
+// if document content length greater, then we'll trigger the summary task
+const MIN_TOKENS_TO_SUMMARIZE = 1000;
+// maximum token length of generated summaries - will truncate if greater
+const MAX_SUMMARY_TOKEN_LENGTH = 1000;
+
+export const retrieveDocumentation =
+  ({
+    outputAPI,
+    searchDocAPI,
+    logger: log,
+  }: {
+    outputAPI: OutputAPI;
+    searchDocAPI: ProductDocSearchAPI;
+    logger: Logger;
+  }): RetrieveDocumentationAPI =>
```

Comment:
So it could be yes, but - and of course we would need proper evaluation scenarios to confirm - the AI generated summaries are "meta summaries" atm, so I don't think it would be very useful.

For example for the "how to create a space" document, the summary is basically "this doc explains how to create a space in Kibana, and list the steps required to do so"